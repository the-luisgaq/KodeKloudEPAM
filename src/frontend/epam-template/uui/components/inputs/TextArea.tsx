import { Overwrite, withMods } from '@epam/uui-core';
import { TextArea as uuiTextArea, TextAreaProps as uuiTextAreaProps } from '@epam/uui-components';
import * as types from '../types';
import { settings } from '../../settings';

import css from './TextArea.module.scss';

const DEFAULT_MODE = types.EditMode.FORM;

type TextAreaMods = types.IHasEditMode & {
    /**
     * Defines component size.
     * @default '36'
     */
    size?: '24' | '30' | '36' | '42' | '48';
};

export interface TextAreaModsOverride {}

function applyTextAreaMods(mods: TextAreaMods) {
    return [
        css.root,
        'uui-textarea',
        'uui-size-' + (mods.size || settings.textArea.sizes.default),
        'uui-control-mode-' + (mods.mode || DEFAULT_MODE),
    ];
}

/** Represents the properties of a TextArea component. */
export interface TextAreaProps extends uuiTextAreaProps, Overwrite<TextAreaMods, TextAreaModsOverride> {}

export const TextArea = withMods<uuiTextAreaProps, TextAreaProps>(
    uuiTextArea,
    applyTextAreaMods,
    (props) => {
        return {
            autoSize: props.mode === types.EditMode.CELL ? true : props.autoSize,
        };
    },
);
