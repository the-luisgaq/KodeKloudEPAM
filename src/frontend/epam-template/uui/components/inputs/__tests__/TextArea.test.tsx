import React from 'react';
import { TextArea } from '../TextArea';
import { render } from '@epam/uui-test-utils';

describe('TextArea', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(<TextArea value="" onValueChange={ jest.fn } />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be rendered with props correctly', () => {
        const { asFragment } = render(<TextArea value="" onValueChange={ jest.fn } placeholder="Type here" size="36" maxLength={ 200 } rows={ 4 } mode="inline" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
