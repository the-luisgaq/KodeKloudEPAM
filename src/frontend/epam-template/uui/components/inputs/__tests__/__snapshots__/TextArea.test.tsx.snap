// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="container uui-input-box root uui-textarea uui-size-36 uui-control-mode-form"
    tabindex="-1"
  >
    <textarea
      class="-clickable noAutoSize uui-input"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea should be rendered with props correctly 1`] = `
<DocumentFragment>
  <div
    class="container uui-input-box root uui-textarea uui-size-36 uui-control-mode-inline"
    tabindex="-1"
  >
    <textarea
      class="-clickable noAutoSize uui-input"
      maxlength="200"
      placeholder="Type here"
      rows="4"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
